impeller/renderer/backend/vulkan/test/mock_vulkan.cc

```diff
@@ -0,0 +1,283 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "impeller/renderer/backend/vulkan/test/mock_vulkan.h"
```

Comment:
TBH I would probably call this a fake instead of a mock, but I think the design is fine? I was more worried about individual test suites constructing mocks on the fly to test specific subsets of impeller - but a reusable fake seems perfectly reasonable to me.

Reply:
I was playing around in a different PR the ability to make it record what was communicated to it.  I think as we expand the test suite when we fix bugs and add tests it may end up looking more and more like what we think of with mocks.